google_books/src/data/apiDataFunctions.js

```diff
@@ -0,0 +1,36 @@
+const basicSearch = async (
+  searchTerm = "",
+  title = "",
+  author = "",
+  publisher = "",
+  resPerPg = 40
+) => {
+  const titleVal = title !== "" ? `+intitle:${title}` : ``;
+  const authorVal = author !== "" ? `+inauthor:${author}` : ``;
+  const publisherVal = publisher !== "" ? `+inpublisher:${publisher}` : ``;
+
+  const response = await fetch(
+    `https://www.googleapis.com/books/v1/volumes?q=${searchTerm}${titleVal}${authorVal}${publisherVal}&maxResults=${resPerPg}`
+  );
+  // image, author, title, description
+  const basicData = await response.json();
+  // show a max 10 pages and warn, change max results to 20 use &startIndex=${page}
+  const pages = Math.ceil(basicData.totalItems / resPerPg);
+
+  const cleanedData = basicData.items?.map((book) => {
```

Comment:
if items is undefined, you can throw an error from this function to indicate to the user that something went wrong